### ISSUE TITLE
Error in validObject(assays) : invalid class “SimpleAssays” object

### ISSUE THREAD

**Issue by lwaldron** [waldronlab/curatedMetagenomicData#283]
**Describe the bug**

I noticed that the following command in Bioconductor 3.16 and cMD 3.6.0 to serialize the FengQ_2015 dataset returns an error and some warnings:

```r
sampleMetadata[sampleMetadata$study_name == "FengQ_2015", ] |>
                     returnSamples("relative_abundance", rownames = "NCBI")
```

```
Error in validObject(assays) : invalid class “SimpleAssays” object: 
    'dim(x)' must return a numeric vector
In addition: Warning messages:
1: Expected 3 pieces. Additional pieces discarded in 1 rows [2]. 
2: package ‘CRCmodel’ is not available for Bioconductor version '3.16'
```

I checked all datasets and saw it all happens with the `YuJ_2015` dataset. See gist below.

``` r
suppressPackageStartupMessages(library(curatedMetagenomicData))

sampleMetadata[sampleMetadata$study_name == "FengQ_2015", ] |>
                     returnSamples("relative_abundance", rownames = "NCBI")
#> snapshotDate(): 2022-10-31
#> Warning: Expected 3 pieces. Additional pieces discarded in 1 rows [2].
#> 
#> $`2021-03-31.FengQ_2015.relative_abundance`
#> dropping rows without rowTree matches:
#>   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Atopobiaceae|g__Olsenella|s__Olsenella_profusa
#>   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Collinsella|s__Collinsella_stercoris
#>   k__Bacteria|p__Actinobacteria|c__Coriobacteriia|o__Coriobacteriales|f__Coriobacteriaceae|g__Enorma|s__[Collinsella]_massiliensis
#>   k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales|f__Bacillales_unclassified|g__Gemella|s__Gemella_bergeri
#>   k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Carnobacteriaceae|g__Granulicatella|s__Granulicatella_elegans
#>   k__Bacteria|p__Firmicutes|c__Clostridia|o__Clostridiales|f__Ruminococcaceae|g__Ruminococcus|s__Ruminococcus_champanellensis
#>   k__Bacteria|p__Firmicutes|c__Erysipelotrichia|o__Erysipelotrichales|f__Erysipelotrichaceae|g__Bulleidia|s__Bulleidia_extructa
#>   k__Bacteria|p__Proteobacteria|c__Betaproteobacteria|o__Burkholderiales|f__Sutterellaceae|g__Sutterella|s__Sutterella_parvirubra
#>   k__Bacteria|p__Synergistetes|c__Synergistia|o__Synergistales|f__Synergistaceae|g__Cloacibacillus|s__Cloacibacillus_evryensis
#> Warning: package 'CRCmodel' is not available for Bioconductor version '3.16'
#> 
#> A version of this package for your version of R might be available elsewhere,
#> see the ideas at
#> https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages
#> Error in validObject(assays): invalid class "SimpleAssays" object: 
#>     'dim(x)' must return a numeric vector
    
allstudies <- unique(sampleMetadata$study_name)
allres <- sapply(allstudies, function(study) {
    message(study)
    try(
        suppressMessages(sampleMetadata[sampleMetadata$study_name == study,] |>
            returnSamples("relative_abundance", rownames = "NCBI"))) |>
        class()
})
#> AsnicarF_2017
#> AsnicarF_2021
#> BackhedF_2015
#> BedarfJR_2017
#> Bengtsson-PalmeJ_2015
#> BritoIL_2016
#> BrooksB_2017
#> Castro-NallarE_2015
#> ChengpingW_2017
#> ChngKR_2016
#> ChuDM_2017
#> CosteaPI_2017
#> DavidLA_2015
#> DeFilippisF_2019
#> DhakanDB_2019
#> FengQ_2015
#> Warning: Expected 3 pieces. Additional pieces discarded in 1 rows [2].
#> Warning: package 'CRCmodel' is not available for Bioconductor version '3.16'
#> 
#> A version of this package for your version of R might be available elsewhere,
#> see the ideas at
#> https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages
#> Error in validObject(assays) : invalid class "SimpleAssays" object: 
#>     'dim(x)' must return a numeric vector
#> FerrettiP_2018
#> FrankelAE_2017
#> GhensiP_2019
#> GopalakrishnanV_2018
#> GuptaA_2019
#> HallAB_2017
#> HanniganGD_2017
#> HansenLBS_2018
#> Heitz-BuschartA_2016
#> HMP_2012
#> HMP_2019_ibdmdb
#> HMP_2019_t2d
#> IaniroG_2022
#> IjazUZ_2017
#> JieZ_2017
#> KarlssonFH_2013
#> KaurK_2020
#> KeohaneDM_2020
#> KieserS_2018
#> KosticAD_2015
#> LassalleF_2017
#> LeChatelierE_2013
#> LeeKA_2022
#> LifeLinesDeep_2016
#> LiJ_2014
#> LiJ_2017
#> LiSS_2016
#> LiuW_2016
#> LokmerA_2019
#> LomanNJ_2013
#> LoombaR_2017
#> LouisS_2016
#> MatsonV_2018
#> MehtaRS_2018
#> MetaCardis_2020_a
#> NagySzakalD_2017
#> NielsenHB_2014
#> Obregon-TitoAJ_2015
#> OhJ_2014
#> OlmMR_2017
#> PasolliE_2019
#> PehrssonE_2016
#> PetersBA_2019
#> QinJ_2012
#> QinN_2014
#> RampelliS_2015
#> RaymondF_2016
#> RosaBA_2018
#> RubelMA_2020
#> SankaranarayananK_2015
#> SchirmerM_2016
#> ShaoY_2019
#> ShiB_2015
#> SmitsSA_2017
#> TettAJ_2016
#> TettAJ_2019_a
#> TettAJ_2019_b
#> TettAJ_2019_c
#> ThomasAM_2018a
#> ThomasAM_2018b
#> ThomasAM_2019_c
#> VatanenT_2016
#> VilaAV_2018
#> VincentC_2016
#> VogtmannE_2016
#> WampachL_2018
#> WindTT_2020
#> WirbelJ_2018
#> XieH_2016
#> YachidaS_2019
#> YassourM_2016
#> YassourM_2018
#> YeZ_2018
#> YuJ_2015
#> Warning: Expected 3 pieces. Additional pieces discarded in 1 rows [2].
#> package 'CRCmodel' is not available for Bioconductor version '3.16'
#> 
#> A version of this package for your version of R might be available elsewhere,
#> see the ideas at
#> https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages
#> Error in validObject(assays) : invalid class "SimpleAssays" object: 
#>     'dim(x)' must return a numeric vector
#> ZeeviD_2015
#> ZellerG_2014
#> ZhuF_2020
names(allres)[allres == "try-error"]
#> [1] "FengQ_2015" "YuJ_2015"
BiocManager::version()
#> [1] '3.16'
BiocManager::valid()
#> 'getOption("repos")' replaces Bioconductor standard repositories, see
#> '?repositories' for details
#> 
#> replacement repositories:
#>     CRAN: https://cloud.r-project.org
#> Warning: 0 packages out-of-date; 1 packages too new
#> 
#> * sessionInfo()
#> 
#> R version 4.2.2 Patched (2022-11-10 r83330)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Ubuntu 20.04.5 LTS
#> 
#> Matrix products: default
#> BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
#> LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
#> 
#> locale:
#>  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#>  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#>  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#>  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#>  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#> [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#> 
#> attached base packages:
#> [1] stats4    stats     graphics  grDevices utils     datasets  methods  
#> [8] base     
#> 
#> other attached packages:
#>  [1] curatedMetagenomicData_3.6.0   TreeSummarizedExperiment_2.6.0
#>  [3] Biostrings_2.66.0              XVector_0.38.0                
#>  [5] SingleCellExperiment_1.20.0    SummarizedExperiment_1.28.0   
#>  [7] Biobase_2.58.0                 GenomicRanges_1.50.1          
#>  [9] GenomeInfoDb_1.34.3            IRanges_2.32.0                
#> [11] S4Vectors_0.36.0               BiocGenerics_0.44.0           
#> [13] MatrixGenerics_1.10.0          matrixStats_0.63.0            
#> 
#> loaded via a namespace (and not attached):
#>   [1] AnnotationHub_3.6.0           BiocFileCache_2.6.0          
#>   [3] plyr_1.8.8                    lazyeval_0.2.2               
#>   [5] splines_4.2.2                 BiocParallel_1.32.1          
#>   [7] ggplot2_3.4.0                 scater_1.26.1                
#>   [9] digest_0.6.30                 yulab.utils_0.0.5            
#>  [11] htmltools_0.5.3               viridis_0.6.2                
#>  [13] fansi_1.0.3                   magrittr_2.0.3               
#>  [15] memoise_2.0.1                 ScaledMatrix_1.6.0           
#>  [17] cluster_2.1.4                 DECIPHER_2.26.0              
#>  [19] R.utils_2.12.2                colorspace_2.0-3             
#>  [21] blob_1.2.3                    rappdirs_0.3.3               
#>  [23] ggrepel_0.9.2                 xfun_0.35                    
#>  [25] dplyr_1.0.10                  crayon_1.5.2                 
#>  [27] RCurl_1.98-1.9                jsonlite_1.8.3               
#>  [29] ape_5.6-2                     glue_1.6.2                   
#>  [31] gtable_0.3.1                  zlibbioc_1.44.0              
#>  [33] DelayedArray_0.24.0           R.cache_0.16.0               
#>  [35] BiocSingular_1.14.0           scales_1.2.1                 
#>  [37] DBI_1.1.3                     Rcpp_1.0.9                   
#>  [39] viridisLite_0.4.1             xtable_1.8-4                 
#>  [41] decontam_1.18.0               tidytree_0.4.1               
#>  [43] bit_4.0.5                     rsvd_1.0.5                   
#>  [45] httr_1.4.4                    ellipsis_0.3.2               
#>  [47] pkgconfig_2.0.3               R.methodsS3_1.8.2            
#>  [49] scuttle_1.8.1                 dbplyr_2.2.1                 
#>  [51] utf8_1.2.2                    tidyselect_1.2.0             
#>  [53] rlang_1.0.6                   reshape2_1.4.4               
#>  [55] later_1.3.0                   AnnotationDbi_1.60.0         
#>  [57] munsell_0.5.0                 BiocVersion_3.16.0           
#>  [59] tools_4.2.2                   cachem_1.0.6                 
#>  [61] cli_3.4.1                     DirichletMultinomial_1.40.0  
#>  [63] generics_0.1.3                RSQLite_2.2.18               
#>  [65] ExperimentHub_2.6.0           mia_1.6.0                    
#>  [67] evaluate_0.18                 stringr_1.4.1                
#>  [69] fastmap_1.1.0                 yaml_2.3.6                   
#>  [71] knitr_1.41                    bit64_4.0.5                  
#>  [73] fs_1.5.2                      purrr_0.3.5                  
#>  [75] KEGGREST_1.38.0               nlme_3.1-160                 
#>  [77] sparseMatrixStats_1.10.0      mime_0.12                    
#>  [79] R.oo_1.25.0                   compiler_4.2.2               
#>  [81] rstudioapi_0.14               beeswarm_0.4.0               
#>  [83] filelock_1.0.2                curl_4.3.3                   
#>  [85] png_0.1-7                     interactiveDisplayBase_1.36.0
#>  [87] reprex_2.0.2                  treeio_1.22.0                
#>  [89] tibble_3.1.8                  stringi_1.7.8                
#>  [91] highr_0.9                     lattice_0.20-45              
#>  [93] Matrix_1.5-3                  styler_1.8.1                 
#>  [95] vegan_2.6-4                   permute_0.9-7                
#>  [97] vctrs_0.5.1                   pillar_1.8.1                 
#>  [99] lifecycle_1.0.3               BiocManager_1.30.19          
#> [101] BiocNeighbors_1.16.0          bitops_1.0-7                 
#> [103] irlba_2.3.5.1                 httpuv_1.6.6                 
#> [105] R6_2.5.1                      promises_1.2.0.1             
#> [107] gridExtra_2.3                 vipor_0.4.5                  
#> [109] codetools_0.2-18              MASS_7.3-58.1                
#> [111] assertthat_0.2.1              withr_2.5.0                  
#> [113] GenomeInfoDbData_1.2.9        mgcv_1.8-41                  
#> [115] parallel_4.2.2                MultiAssayExperiment_1.24.0  
#> [117] grid_4.2.2                    beachmat_2.14.0              
#> [119] tidyr_1.2.1                   rmarkdown_2.18               
#> [121] DelayedMatrixStats_1.20.0     shiny_1.7.3                  
#> [123] ggbeeswarm_0.6.0             
#> 
#> Bioconductor version '3.16'
#> 
#>   * 0 packages out-of-date
#>   * 1 packages too new
#> 
#> create a valid installation with
#> 
#>   BiocManager::install("cBioPortalData", update = TRUE, ask = FALSE, force = TRUE)
#> 
#> more details: BiocManager::valid()$too_new, BiocManager::valid()$out_of_date
sessionInfo()
#> R version 4.2.2 Patched (2022-11-10 r83330)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Ubuntu 20.04.5 LTS
#> 
#> Matrix products: default
#> BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
#> LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
#> 
#> locale:
#>  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#>  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#>  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#>  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#>  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#> [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#> 
#> attached base packages:
#> [1] stats4    stats     graphics  grDevices utils     datasets  methods  
#> [8] base     
#> 
#> other attached packages:
#>  [1] curatedMetagenomicData_3.6.0   TreeSummarizedExperiment_2.6.0
#>  [3] Biostrings_2.66.0              XVector_0.38.0                
#>  [5] SingleCellExperiment_1.20.0    SummarizedExperiment_1.28.0   
#>  [7] Biobase_2.58.0                 GenomicRanges_1.50.1          
#>  [9] GenomeInfoDb_1.34.3            IRanges_2.32.0                
#> [11] S4Vectors_0.36.0               BiocGenerics_0.44.0           
#> [13] MatrixGenerics_1.10.0          matrixStats_0.63.0            
#> 
#> loaded via a namespace (and not attached):
#>   [1] AnnotationHub_3.6.0           BiocFileCache_2.6.0          
#>   [3] plyr_1.8.8                    lazyeval_0.2.2               
#>   [5] splines_4.2.2                 BiocParallel_1.32.1          
#>   [7] ggplot2_3.4.0                 scater_1.26.1                
#>   [9] digest_0.6.30                 yulab.utils_0.0.5            
#>  [11] htmltools_0.5.3               viridis_0.6.2                
#>  [13] fansi_1.0.3                   magrittr_2.0.3               
#>  [15] memoise_2.0.1                 ScaledMatrix_1.6.0           
#>  [17] cluster_2.1.4                 DECIPHER_2.26.0              
#>  [19] R.utils_2.12.2                colorspace_2.0-3             
#>  [21] blob_1.2.3                    rappdirs_0.3.3               
#>  [23] ggrepel_0.9.2                 xfun_0.35                    
#>  [25] dplyr_1.0.10                  crayon_1.5.2                 
#>  [27] RCurl_1.98-1.9                jsonlite_1.8.3               
#>  [29] ape_5.6-2                     glue_1.6.2                   
#>  [31] gtable_0.3.1                  zlibbioc_1.44.0              
#>  [33] DelayedArray_0.24.0           R.cache_0.16.0               
#>  [35] BiocSingular_1.14.0           scales_1.2.1                 
#>  [37] DBI_1.1.3                     Rcpp_1.0.9                   
#>  [39] viridisLite_0.4.1             xtable_1.8-4                 
#>  [41] decontam_1.18.0               tidytree_0.4.1               
#>  [43] bit_4.0.5                     rsvd_1.0.5                   
#>  [45] httr_1.4.4                    ellipsis_0.3.2               
#>  [47] pkgconfig_2.0.3               R.methodsS3_1.8.2            
#>  [49] scuttle_1.8.1                 dbplyr_2.2.1                 
#>  [51] utf8_1.2.2                    tidyselect_1.2.0             
#>  [53] rlang_1.0.6                   reshape2_1.4.4               
#>  [55] later_1.3.0                   AnnotationDbi_1.60.0         
#>  [57] munsell_0.5.0                 BiocVersion_3.16.0           
#>  [59] tools_4.2.2                   cachem_1.0.6                 
#>  [61] cli_3.4.1                     DirichletMultinomial_1.40.0  
#>  [63] generics_0.1.3                RSQLite_2.2.18               
#>  [65] ExperimentHub_2.6.0           mia_1.6.0                    
#>  [67] evaluate_0.18                 stringr_1.4.1                
#>  [69] fastmap_1.1.0                 yaml_2.3.6                   
#>  [71] knitr_1.41                    bit64_4.0.5                  
#>  [73] fs_1.5.2                      purrr_0.3.5                  
#>  [75] KEGGREST_1.38.0               nlme_3.1-160                 
#>  [77] sparseMatrixStats_1.10.0      mime_0.12                    
#>  [79] R.oo_1.25.0                   compiler_4.2.2               
#>  [81] rstudioapi_0.14               beeswarm_0.4.0               
#>  [83] filelock_1.0.2                curl_4.3.3                   
#>  [85] png_0.1-7                     interactiveDisplayBase_1.36.0
#>  [87] reprex_2.0.2                  treeio_1.22.0                
#>  [89] tibble_3.1.8                  stringi_1.7.8                
#>  [91] highr_0.9                     lattice_0.20-45              
#>  [93] Matrix_1.5-3                  styler_1.8.1                 
#>  [95] vegan_2.6-4                   permute_0.9-7                
#>  [97] vctrs_0.5.1                   pillar_1.8.1                 
#>  [99] lifecycle_1.0.3               BiocManager_1.30.19          
#> [101] BiocNeighbors_1.16.0          bitops_1.0-7                 
#> [103] irlba_2.3.5.1                 httpuv_1.6.6                 
#> [105] R6_2.5.1                      promises_1.2.0.1             
#> [107] gridExtra_2.3                 vipor_0.4.5                  
#> [109] codetools_0.2-18              MASS_7.3-58.1                
#> [111] assertthat_0.2.1              withr_2.5.0                  
#> [113] GenomeInfoDbData_1.2.9        mgcv_1.8-41                  
#> [115] parallel_4.2.2                MultiAssayExperiment_1.24.0  
#> [117] grid_4.2.2                    beachmat_2.14.0              
#> [119] tidyr_1.2.1                   rmarkdown_2.18               
#> [121] DelayedMatrixStats_1.20.0     shiny_1.7.3                  
#> [123] ggbeeswarm_0.6.0
```

<sup>Created on 2022-11-23 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**To Reproduce**

https://gist.github.com/lwaldron/aca4b184183b15c2686f26a5508e52ee

**Expected behavior**

`returnSamples()` should return a (Tree)SummarizedExperiment

**Additional context**

I have not tested other objects besides relative_abundance so there may be invalid objects there too.

**Comment by lwaldron** [comment]
FYI @haoyanzh - if you remove for now FengQ_2015 and YuJ_2015 you should be fine.

**Comment by schifferl** [comment]
Thanks for the report @lwaldron, I will have a look and start working on a patch.

**Comment by schifferl** [comment]
Hi @lwaldron, this was caused because resources in ExperimentHub for the `CRCmodel` package (perhaps unreleased because I can't find it) match the names of resources from `curatedMetagenomicData` exactly (i.e. they use *our* names for *their* resources). The issue was resolved by 01553e6 in release and a8229bf in devel; the changes have been pushed to Bioconductor and will be built in the next few days.
